src/DGtal/shapes/Mesh.ih

```diff
@@ -285,6 +286,29 @@ DGtal::Mesh<TPoint>::getFaceColor(unsigned int i) const
 
 template<typename TPoint>
 inline
+std::pair<TPoint, TPoint>  
+DGtal::Mesh<TPoint>::getBoundingBox() const
+{
+  std::pair<TPoint, TPoint> theResult;  
+  struct CompPoints
+  {
+    CompPoints(typename TPoint::Dimension d): myDim(d){};
+    bool operator() (const TPoint &p1, const TPoint &p2){return p1[myDim]<p2[myDim];};
+    typename TPoint::Dimension myDim;
+  };
+  TPoint lowerBound, upperBound;
+  for(unsigned int i=0; i< TPoint::size(); i++)
+    {
+      upperBound[i] = (*(std::max_element(vertexBegin(), vertexEnd(), CompPoints(i))))[i];
+      lowerBound[i] = (*(std::min_element(vertexBegin(), vertexEnd(), CompPoints(i))))[i];
```

Comment:
Ho, my bad, I just seen that boost::bind overloads `<` ... that greatly simplified the expression :smile:
```C++
upperBound[i] = (*( std::max_element( vertexBegin(), vertexEnd(), 
  boost::bind( static_cast<typename TPoint::Component const& (TPoint::*)(typename TPoint::Dimension) const >(&TPoint::operator[]), _1, i )
  < boost::bind( static_cast<typename TPoint::Component const& (TPoint::*)(typename TPoint::Dimension) const >(&TPoint::operator[]), _2, i )
) ))[i];
```